lib/aptible/cli/subcommands/db.rb

```diff
@@ -113,14 +113,15 @@ def self.included(thor)
               render_database(database, database.account)
             end
 
-            desc 'db:dump HANDLE', 'Dump a remote database to file'
+            desc 'db:dump HANDLE [pg_dump options]',
+                 'Dump a remote database to file'
             option :environment
-            define_method 'db:dump' do |handle|
+            define_method 'db:dump' do |handle, *dump_options|
               database = ensure_database(options.merge(db: handle))
               with_postgres_tunnel(database) do |url|
                 filename = "#{handle}.dump"
                 CLI.logger.info "Dumping to #{filename}"
-                `pg_dump #{url} > #{filename}`
+                `pg_dump #{url} #{dump_options.join(' ')} > #{filename}`
```

Comment:
Passing through uncontrolled options to a shell like this is problematic from a security perspective - e.g. you could pass `aptible db:dump '&& echo oops'` and `echo oops` would end up being executed (which is neither desirable nor expected).

To fix the issue, use [`Array#shelljoin`](http://ruby-doc.org/stdlib-2.0.0/libdoc/shellwords/rdoc/Shellwords.html#method-c-shelljoin) rather than `Array#join`:

```ruby
`pg_dump #{url} #{dump_options.shelljoin} > #{filename}`
```

Reply:
Done